lms/static/js/learner_dashboard/models/program_subscription_model.js

```diff
@@ -0,0 +1,81 @@
+import Backbone from 'backbone';
+import moment from 'moment';
+
+import DateUtils from 'edx-ui-toolkit/js/utils/date-utils';
+
+/**
+ * Model for Program Subscription Data.
+ */
+class ProgramSubscriptionModel extends Backbone.Model {
+    constructor([data = {}], urls, userPreferences, ...args) {
+        const subscriptionState = data.subscription_state?.toLowerCase();
+        const subscriptionPrice =
+            data.currency === 'USD'
+                ? `$${data.price}`
+                : `${data.price} ${data.currency}`;
+        const subscriptionUrl =
+            subscriptionState === 'active'
+                ? urls.manage_subscription_url
+                : urls.buy_subscription_url;
+
+        const hasActiveTrial =
+            subscriptionState === 'active' && data.trial_end
+                ? moment(
+                    DateUtils.localizeTime(
+                        DateUtils.stringToMoment(data.trial_end),
+                        'UTC'
+                    )
+                ).isAfter(moment.utc())
+                : false;
+
+        const [nextPaymentDate] = ProgramSubscriptionModel.formatDate(
+            data.next_payment_date,
+            userPreferences
+        );
+        const [trialEndDate, trialEndTime] = ProgramSubscriptionModel.formatDate(
+            data.trial_end,
+            userPreferences
+        );
+
+        const trialLength = 7;
+
+        super(
+            {
+                hasActiveTrial,
+                nextPaymentDate,
+                subscriptionPrice,
+                subscriptionState,
+                subscriptionUrl,
+                trialEndDate,
+                trialEndTime,
+                trialLength,
+            },
+            ...args
+        );
+    }
+
+    static formatDate(date, userPreferences) {
```

Comment:
I think this should be a utility function not part of backbone modal.

Reply:
I noticed a similar convention in existing models, so I followed it [Similar Example](https://github.com/openedx/edx-platform/blob/5549db4d80dfc80ab90d529346c7c3efc9b801ee/lms/static/js/learner_dashboard/models/course_card_model.js#L105-L119)